File: mobile/lib/models/doctor.dart
class Doctor {
  final String id;
  final String name;
  final String email;
  final String role;
  final String department;

  Doctor({
    required this.id,
    required this.name,
    required this.email,
    required this.role,
    required this.department,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      role: json['role'] ?? 'Doctor',
      department: json['department'] ?? '',
    );
  }
}

File: backend/src/models/Doctor.js
const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema({
  name: { type: String, required: true },
  email: { type: String, required: true, unique: true },
  passwordHash: { type: String, required: true },
  role: { type: String, enum: ['doctor', 'admin'], default: 'doctor' },
  department: String,
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: 'Hospital' },
  licenseNo: String
}, { timestamps: true });

module.exports = mongoose.model('Doctor', doctorSchema);
